sync_labels.py

```diff
@@ -0,0 +1,148 @@
+    """
+
+    # Get PyGithub objects representing repos
+    g = Github(get_access_token())
+    try:
+        s = g.get_repo(src_repo)
+    except GithubException:
+        raise Exception(f"Source repo is not a repo: {src_repo}")
+    try:
+        d = g.get_repo(dest_repo)
+    except GithubException:
+        raise Exception(f"Destination repo is not a repo: {dest_repo}")
+
+    # Create list of labels to add/delete
+    src_labels = list(s.get_labels())
+    src_label_names = set([j.name for j in src_labels])
+    dest_labels = list(d.get_labels())
+    dest_label_names = set([j.name for j in dest_labels])
+    dest_add = src_label_names - dest_label_names
+    dest_del = dest_label_names - src_label_names
+
+    # Show summary of what actions will be taken, get confirmation
+    print(f"The following labels will be created in {dest_repo}:")
+    for lab in sorted(list(dest_add)):
+        print(f" [+] {lab}")
+
+    if delete:
+        print(f"The following labels will be deleted in {dest_repo}:")
+        for lab in sorted(list(dest_del)):
+            print(f" [-] {lab}")
+
+    # Add new labels
+    for name in dest_add:
+        print(f"Creating label {name} in destination repo {dest_repo}...")
+        if dry_run:
+            print(f"Dry run successful!")
+            continue
+        label = s.get_label(name)
+        color = label.color
+        description = label.description
+        try:
+            d.create_label(name, color, description)
+            print(f"Success!")
+        except GithubException:
+            raise Exception(f"Error creating label {name} in destination repo {dest_repo}")
+
+    # Delete labels
+    if delete:
+        for name in dest_del:
+            print(f"Removing label {name} from destination repo {dest_repo}...")
+            if dry_run:
+                print(f"Dry run successful!")
+                continue
+            label = d.get_label(name)
+            try:
+                label.delete()
+                print(f"Success!")
+            except GithubException:
+                raise Exception(f"Error deleting label {name} from destination repo {dest_repo}")
+
+    print("Done.")
+
+
+def get_access_token():
+    """Grab github api token from env var"""
+    api_key = os.environ.get("GITHUB_API_KEY", None)
+    if api_key is None:
+        raise Exception("No env var GITHUB_API_KEY defined")
+    return api_key
+
+
+def get_user_confirmation():
+    """Make sure user is ok to proceed"""
+    confirm = "Okay to proceed? (Type 'y' or 'yes' to confirm):\n"
+    response = input(confirm)
+    if response.lower() not in ["y", "yes"]:
+        print("You safely aborted the label sync operation.")
+        sys.exit(0)
+
+
+def main(sysargs=sys.argv[1:]):
+
+    parser = get_argument_parser(sysargs)
+
+    # Make sure the user provided SOME arguments
+    if len(sys.argv) == 1:
+        parser.print_help(sys.stderr)
+        sys.exit(1)
```

Comment:
This should be unnecessary if you're using `argparse` correctly.

Reply:
addressed by f063749